src/bindings/jna/stdioproc/src/main/java/org/libelektra/stdioproc/StdIoProc.java

```diff
@@ -0,0 +1,194 @@
+package org.libelektra.stdioproc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import org.libelektra.KDBException;
+import org.libelektra.Key;
+import org.libelektra.KeySet;
+import org.libelektra.Plugin;
+
+class StdIoProc {
+
+  private static final String STATUS_SUCCESS = "success";
+  private static final String STATUS_NO_UPDATE = "noupdate";
+  private static final String STATUS_ERROR = "error";
+
+  private final Plugin plugin;
+  private final Dump dump;
+
+  final InputStream inputStream;
+  final OutputStream outputStream;
+
+  StdIoProc(Plugin plugin, InputStream inputStream, OutputStream outputStream) {
+    this.plugin = plugin;
+    this.inputStream = inputStream;
+    this.outputStream = outputStream;
+    this.dump = new Dump(this);
+  }
+
+  public boolean handshake() throws IOException {
+    var header = readLine();
+    if (!Objects.equals(header, "ELEKTRA_STDIOPROC INIT v1")) {
+      return false;
+    }
+
+    println("ELEKTRA_STDIOPROC ACK v1");
+    println(plugin.getName());
+
+    var jniContract = KeySet.create();
+    try {
+      plugin.get(jniContract, Key.create(Plugin.JNI_MODULE_CONTRACT_ROOT));
+    } catch (KDBException e) {
+      // ignored
+    }
+
+    var contract = KeySet.create();
+    for (Key key : jniContract) {
+      Key k = key.dup();
+      k.setName(
+          key.getName()
+              .replace(Plugin.JNI_MODULE_CONTRACT_ROOT, "system:/elektra/modules/stdioproc"));
+      contract.append(k);
+    }
+
+    contract.append(Key.create("system:/elektra/modules/stdioproc/exports/open", "1"));
+    contract.append(Key.create("system:/elektra/modules/stdioproc/exports/get", "1"));
+    contract.append(Key.create("system:/elektra/modules/stdioproc/exports/set", "1"));
+    contract.append(Key.create("system:/elektra/modules/stdioproc/exports/close", "1"));
```

Comment:
I really don't like this bit. Adding at least a `getContract()` would be much better (even better would be also adding `getSupportedOperations()`).

Reply:
It is/was planned with the new backend. Implementing the change separately would be a lot of unnecessary overhead.